src/routers/xcm/Router.sol

```diff
@@ -44,6 +69,11 @@ contract ConnectorXCMRouter {
         _;
     }
 
+    // todo(nuno): add auth modifier
+    function update(XcmWeightInfo memory xcmWeightInfo_) external {
+        xcmWeightInfo = xcmWeightInfo_;
+    }
```

Comment:
@offerijns @AStox I don't know what's the Solidity way of doing this. The `file` pattern doesn't really seem applicable here because we are not handling random `bytes` data as input.

Reply:
I think we can still use the `file` pattern, just
```solidity
 function file(bytes32 what, XcmWeightInfo memory xcmWeightInfo_) external auth {
        if (what == "weight-info") xcmWeightInfo = xcmWeightInfo_;
        else revert("XCMRouter/file-unrecognized-param");
        emit File(what, data);
 }
```

Reply:
I don't get that tho. Why having a "what" if it clearly only works to set the XcmWeightInfo?

Reply:
This is most likely going to be called via multi-sig, and I'm not sure how well 3rd party UIs like gnosis safe or etherscan will parse structs. In Tinlake, for things like rate groups, we pass in all the parameters individually and instantiate a new struct.

```solidity
function file(bytes32 what, uint64 buyExecutionWeightLimit, uint64 transactWeightAtMost, uint256 feeAmount) external auth { 
    if (what == "weight-info") {
        xcmWeightInfo = xcmWeightInfo(buyExecutionWeightLimit, transactWeightAtMost, feeAmount);
    } else {
        revert("XCMRouter/file-unrecognized-param"); 
    }
    emit File(what, data); 
}
```

Reply:
Good insight, @AStox. thanks! I guess the only purpose of `what` here is to have that identifying the `File` event? But then again, in your example you file `data` where that doesn't apply in this case.

I understand this is a Solidity idiomatic issue but I am not quite the fan of perpetuating unreasonable designs however idiomatic they are. If the `File` event pattern is something that other tools will be expecting and we should therefore comply with, that's fine I guess. Let me know 👍

Reply:
FYI @NunoAlexandre https://mirror.xyz/horsefacts.eth/R5N_Dzm2XKVvSI3cM8e8EHuFzvpPBttG2TEtyEZDa10

I think the benefit of having a single method name and single event name that can be audited for parameter updates is quite significant.

See also https://github.com/makerdao/dss-direct-deposit/blob/master/src/plans/D3MAavePlan.sol#L110 for some more recent code that uses the same pattern in what I consider to be clean, easily auditable Solidity code (there's definitely some subjectivity here).

Reply:
Thanks a ton for the context, @offerijns 🙏 

Done 👍